src/SavingIterator.php

```diff
@@ -41,16 +42,20 @@ public function __construct(
         private Iterator $origin,
         AddingIterator $target
     ) {
+        $methods = array_flip(["current", "key"]);
         /** @phpstan-ignore-next-line */
         $this->target = new ContextVeil(
             $target,
-            fn (AddingIterator $stored) =>
-            ($this->origin->valid())
-            ? $stored->from(
-                $this->origin
+            new ClosureReaction(
+                fn (AddingIterator $stored, string $method) => (
+                    ($this->origin->valid())
+                    && (isset(
+                        $methods[$method]
+                    ))
+                ) ? $stored->from(
+                    $this->origin
+                ) : $stored
```

Comment:
Technically this is a mess. If the origin `Iterator` still has new elements and currently called method is one of the methods which have to be reacted to then a new element is retrieved.